gaphor/SysML/blocks/connectors.py

```diff
@@ -38,17 +38,17 @@ def connect(self, handle: Handle, port: Port) -> bool:
             proxy_port.subject.encapsulatedClassifier = self.block.subject
 
         # This raises the item in the item hierarchy
-        assert proxy_port.canvas
-        proxy_port.canvas.reparent(proxy_port, self.block)
+        assert proxy_port.diagram
+        proxy_port.parent = self.block
 
         return True
 
     def disconnect(self, handle: Handle) -> None:
         proxy_port = self.proxy_port
-        if proxy_port.subject and proxy_port.canvas:
+        if proxy_port.subject and proxy_port.diagram:
             subject = proxy_port.subject
             del proxy_port.subject
-            proxy_port.canvas.reparent(proxy_port, None)
+            proxy_port.parent = None  # type: ignore[assignment]
```

Comment:
I really like the simplifications here and throughout the refactoring to be able to call `.parent` :+1: